Sources/XCHammer/XcodeBuildSystemInstaller.swift

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019-present, Pinterest, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import Result
+
+enum XcodeBuildSystemInstaller {
+    /// Installs the Xcode build system contained inside the bundle if necessary
+    /// It noops quickly if the installed plist doesnt match in order to be ran
+    /// inline with builds or project generation
+    static func installIfNecessary() -> Result<(), CommandError> {
+        let bundle = Bundle.main
+        let buildkitBundlePath = bundle.path(forResource: "XCBuildKit", ofType: "bundle")!
+        let plistPath = buildkitBundlePath + "/BuildInfo.plist"
+        let installedPath = "/opt/XCBuildKit/XCBuildKit.app/Contents/Info.plist"
+        if getVersion(path: plistPath) != getVersion(path: installedPath) {
+            let installerPath = buildkitBundlePath + "/BazelBuildServiceInstaller.pkg"
+            let script = "installer -pkg \(installerPath) -target /"
+            guard ShellOutWithSudo(script) == 0 else {
+                return .failure(.basic("failed to install"))
+            }
+        }
+        return .success(())
+    }
+
+    static func getVersion(path: String) -> String? {
+        guard let plistXML = FileManager.default.contents(atPath: path) else {
+            return nil
+        }
+        var propertyListFormat =  PropertyListSerialization.PropertyListFormat.xml
+        guard let plistData: [String: AnyObject] = try? PropertyListSerialization.propertyList(from:
+            plistXML, options: .mutableContainersAndLeaves,
+            format: &propertyListFormat) as! [String:AnyObject] else {
+            fatalError("Can't read plist")
+        }
+        return plistData["BUILD_COMMIT"] as? String
```

Comment:
Can you add a link to where this is added to the Info.plist in `xcbuildkit`?

Reply:
Done